### ISSUE TITLE
Fix typing lag in chat + option to clear bans without restarting room

### ISSUE THREAD

**Issue by SpiderSuave** [m1k1o/neko#356]
Hello and as always thank you for your work.

I've noticed a typing lag problem that Neko has always had, where when the chatroom gets full of too much text, the user's typing starts to lag and they must refresh their web page to fix the problem.

Also an option to clear bans without restarting the room would be awesome. Or perhaps someone could let me know of a way to do this from the server side where Neko is executed.

Thanks again.

**Comment by m1k1o** [comment]
I wonder how many chat messages are you talking about, i never saw such issue. But you are right, those messages are never deleted until the page is refreshed.

**Comment by SpiderSuave** [comment]
When there's enough scrollable text in the chatroom filling roughly about three screens, the lag begins, a little at first and it gets progressively worse, like there's some kind of memory leak happening.

It happens to me and I've confirmed it with others when I host watchparties, and it doesn't seem to make a difference how many people are in the room, whether it's one or a dozen or more people.

Thank you for any help on this.

**Comment by m1k1o** [comment]
Yeah, that lag must be only on the client side because no chat is stored or processed on the server. We could remove old messages, but that would be inconvenient for people that want to browse session history. Or we could only display messages that are actually visible in user's viewport and save others to some variable. But that is quite a lot of work.

**Comment by mbattista** [comment]
I will try to wrap the chat into a virtual scroller instead of a v-for over history. This should fix it, but would add a dependency.

**Comment by SpiderSuave** [comment]
Thank you, or maybe just an option to set a limit for text history that clients can scroll through, unless that's what you mean. It's a problem caused on the client side yes, since anyone simply reloading the page fixes it for them until text accumulates again to the point of lag. As always, much appreciated. I love this app and love to see it improved and you guys are great.

**Comment by SpiderSuave** [comment]
Just to note, it's not a matter of total text, but of number of chat messages sent, number of entries, and it's about at 300 that the lag starts to become noticeable, or at least noticeable enough.

In other words, if I paste 10 lines per chat message, about 10 times more text can be in the chatroom before lag starts getting noticeable, but with just even one character per chat message, at 300 or so entries it will be lagging.

Thanks again.
